units_py/resources.py

```diff
@@ -1,33 +1,68 @@
+"""Defines Resources, an object for managing a resource set."""
+
+
 import attr
 
+
 @attr.s
 class Resources:
+    """@brief An object for managing a resource set."""
     red: int   = attr.ib(default = 0)
     green: int = attr.ib(default = 0)
     blue: int  = attr.ib(default = 0)
 
-    # Creates resources from string formatted like '{red} {green} {blue}'.
-    # Throws if string is invalid.
+
     @staticmethod
     def of_string_exn(s: str) -> 'Resources':
+        """@brief Creates and returns Resources object from a string formated like '{red} {green} {blue}'.
+
+        @param s: The input string.
+
+        @return Resources object formed from s.
+        """
         red, green, blue = map(int, s.split(' '))
+
         return Resources(red=red, green=green, blue=blue)
-    
+
+
     def to_string_hum(self) -> str:
-        return '((red {}) (green {}) (blue {})'.format(self.red, self.green, self.blue)
-    
-    # Whether [self] can afford the cost represented by [other]
+        """@brief Returns a human-readable string describing this object.
+
+        @return string "((red <red count>) (green <green count>) (blue <blue count>))"
+        """
+
+        return '((red {red}) (green {green}) (blue {blue})'.format(red=self.red, green=self.green, blue=self.blue)
+
+
     def subsumes(self, other: 'Resources') -> bool:
+        """@brief Returns whether the parameter 'other' represents more resources than are in this object.
```

Comment:
This is backwards. `x.subsumes(y)` iff x has more resources. Could rename to `affords` maybe.

Reply:
`affords` is a good name. Aside: It's a non-strict equality, so it should be "at least as many" rather than "more".